s3auth-hosts/src/main/java/com/s3auth/hosts/DefaultDynamo.java

```diff
@@ -200,35 +205,74 @@ public void close() {
-        attrs.put(DefaultDynamo.KEY, new AttributeValue(domain.key()));
-        attrs.put(DefaultDynamo.SECRET, new AttributeValue(domain.secret()));
-        attrs.put(DefaultDynamo.REGION, new AttributeValue(domain.region()));
-        attrs.put(DefaultDynamo.SYSLOG, new AttributeValue(domain.syslog()));
-        attrs.put(DefaultDynamo.BUCKET, new AttributeValue(domain.bucket()));
-        final AmazonDynamoDB amazon = this.client.get();
-        amazon.putItem(new PutItemRequest(this.table, attrs));
-        amazon.shutdown();
-        return true;
+        final Region region = this.regionfactory.createRegion();
+        final Table curTable = region.table(this.table);
+        boolean success = false;
+        try {
+            curTable.put(new Attributes()
+                .with(
+                    DefaultDynamo.USER,
+                    new AttributeValue(
+                        user.toString()
+                    )
+            )
+                .with(
+                    DefaultDynamo.NAME,
+                    new AttributeValue(
+                        domain.name()
+                    )
+                )
+                .with(
+                    DefaultDynamo.KEY,
+                    new AttributeValue(
+                        domain.key()
+                    )
+                )
+                .with(
+                    DefaultDynamo.SECRET,
+                    new AttributeValue(
+                        domain.secret()
+                    )
+                )
+                .with(
+                    DefaultDynamo.REGION,
+                    new AttributeValue(
+                        domain.region()
+                    )
+                )
+                .with(
+                    DefaultDynamo.SYSLOG,
+                    new AttributeValue(
+                        domain.syslog()
+                    )
+                )
+                .with(
+                    DefaultDynamo.BUCKET,
+                    new AttributeValue(
+                        domain.bucket()
+                    )
+                )
+            );
+            success = true;
+        } catch (final IOException exception) {
+            Logger.error(this, exception.getMessage());
+        }
+        return success;
     }
 
     @Override
     @Cacheable.FlushBefore
     public boolean remove(@NotNull final Domain domain) {
-        final AmazonDynamoDB amazon = this.client.get();
-        amazon.deleteItem(
-            new DeleteItemRequest(
-                this.table,
-                new ImmutableMap.Builder<String, AttributeValue>()
-                    .put(DefaultDynamo.NAME, new AttributeValue(domain.name()))
-                    .build()
-            )
-        );
-        amazon.shutdown();
+        final Region region = this.regionfactory.createRegion();
+        final Table curTable = region.table(this.table);
+        final Iterator<Item> itemsToRemove = curTable.frame().where(
```

Comment:
Same here, avoid composite variable names: http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html